src/foam/nanos/auth/CachingAuthService.java

```diff
@@ -66,7 +68,16 @@ public void eof() {
         public void reset(Detachable sub) {
           purgeCache(x);
         }
-      }, TRUE);
+      };
+
+      DAO userDAO       = (DAO) x.get("userDAO");
+      DAO groupDAO      = (DAO) x.get("groupDAO");
+      DAO groupPermissionJunctionDAO = (DAO) x.get("groupPermissionJunctionDAO");
+
+      groupDAO.listen(purgeSink, TRUE);
+      userDAO.listen(purgeSink, EQ(User.ID, user.getId()));
```

Comment:
We only need to be purging a user's cache if their `enabled` property changes, not on every put. Same applies to when `Group`s change in `groupDAO`. Clearing on every put to `userDAO` and `groupDAO` could result in far more cache misses than is necessary.

Reply:
We would need lots of logic to determine when we actually need to purge the cache, but that is error prone and since users and groups shouldn't change that often, it isn't much overhead to purge caches and rebuild caches.